### PR TITLE
Mark unfixed clippy warnings as allowed

### PR DESCRIPTION
**Pull request by ipuustin**
The idea is that we'll catch new errors in CI and also we can fix the existing warnings whenever we like.

### CODE REVIEW

**Comment by ipuustin** [comment]
@Mossaka what do you think?

File: crates/containerd-shim-wasmedge/src/instance.rs [hunk 1]
```diff
@@ -135,6 +135,7 @@ pub fn prepare_module(
     if stdin.is_some() {
         unsafe {
             STDIN_FD = Some(dup(STDIN_FILENO));
+            #[allow(clippy::unnecessary_unwrap)]
```
Comment (Mossaka) [inline crates/containerd-shim-wasmedge/src/instance.rs]: I would suggest that we remove `unwrap()` instead of allowing unncessary_unwrap policy.

**Comment by Mossaka** [comment]
Thank you for your contribution @ipuustin . Mostly look good to me. Since we are holding off the commits / PRs to runwasi before merging youki change, do you think we can mostly resovle the clippy issues instead of allowing them in the code?

**Comment by ipuustin** [comment]
Ok! I fixed a bunch of clippy issues. Left unfixed are only the ones in generated files. I think we need to rebase the wasmedge file after the libcontainer integration PR lands.

**Pull request merged**